### DISCUSSION TITLE
Android Auto

### DISCUSSION THREAD

**Discussion by nitrag** [maplibre/maplibre-native#988] [General]
It appears that Android Auto has a more complex system for rendering a map than iOS. In iOS, I have implemented another MapView via ViewController and it was quite easy. I have seen the company that this repository is forked from has a plugin to assist with rendering Android Auto but only compatible with v10+. I'm not talking about Navigation SDK.

- Are there any pre-requisites in maplibre-gl-native that do not exist for AA functionality?
  - If not, does anyone have a working example?
  - If so, what would be the bounty on this?

**Comment by safe-bug** [comment]
@nitrag Do you have any update on this? I want to know if there's an implementation for Android Auto using MapLibre

**Reply by boldtrn** [reply]
There are implementations, but nothing open source as far as I know. For example Flitsmeister has Android Auto working with MapLibre.

**Comment by nitrag** [comment]
Android unfortunately doesn't have the same support as Apple for View Controllers in Car Play. You essentially have to take bitmap renders of a virtual map view and send those at 30fps to Android Screen renderer. 

We spent over $20k in closed source development efforts to fully build this out.
